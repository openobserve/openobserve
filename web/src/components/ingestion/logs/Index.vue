<!-- Copyright 2023 OpenObserve Inc.

This program is free software: you can redistribute it and/or modify
it under the terms of the GNU Affero General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.

This program is distributed in the hope that it will be useful
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU Affero General Public License for more details.

You should have received a copy of the GNU Affero General Public License
along with this program.  If not, see <http://www.gnu.org/licenses/>.
-->

<!-- eslint-disable vue/x-invalid-end-tag -->
<template>
  <q-splitter
    v-model="splitterModel"
    unit="px"
    style="min-height: calc(100vh - 130px)"
  >
    <template v-slot:before>
      <q-tabs
        v-model="ingestiontabs"
        indicator-color="transparent"
        inline-label
        vertical
      >
        <q-route-tab
          name="curl"
          :to="{
            name: 'curl',
            query: {
              org_identifier: store.state.selectedOrganization.identifier,
            },
          }"
          icon="data_object"
          label="Curl"
          content-class="tab_content"
        />
        <q-route-tab
          name="filebeat"
          :to="{
            name: 'filebeat',
            query: {
              org_identifier: store.state.selectedOrganization.identifier,
            },
          }"
          :icon="'img:' + getImageURL('images/ingestion/filebeat.png')"
          label="Filebeat"
          content-class="tab_content"
        />
        <q-route-tab
          default
          name="fluentbit"
          :to="{
            name: 'fluentbit',
            query: {
              org_identifier: store.state.selectedOrganization.identifier,
            },
          }"
          :icon="'img:' + getImageURL('images/ingestion/fluentbit_icon.png')"
          label="FluentBit"
          content-class="tab_content"
        />
        <q-route-tab
          name="fluentd"
          :to="{
            name: 'fluentd',
            query: {
              org_identifier: store.state.selectedOrganization.identifier,
            },
          }"
          :icon="'img:' + getImageURL('images/ingestion/fluentd_icon.svg')"
          label="Fluentd"
          content-class="tab_content"
        />
        <q-route-tab
          name="vector"
          :to="{
            name: 'vector',
            query: {
              org_identifier: store.state.selectedOrganization.identifier,
            },
          }"
          :icon="'img:' + getImageURL('images/ingestion/vector.png')"
          label="Vector"
          content-class="tab_content"
        />
        <q-route-tab
          name="kinesisfirehose"
          :to="{
            name: 'kinesisfirehose',
            query: {
              org_identifier: store.state.selectedOrganization.identifier,
            },
          }"
          :icon="'img:' + getImageURL('images/ingestion/kinesis_firehose.svg')"
          label="Kinesis Firehose"
          content-class="tab_content"
        />
        <q-route-tab
          name="gcpLogs"
          :to="{
            name: 'gcpLogs',
            query: {
              org_identifier: store.state.selectedOrganization.identifier,
            },
          }"
          :icon="'img:' + getImageURL('images/ingestion/gcp.svg')"
          label="GCP Logs via (Pub/Sub)"
          content-class="tab_content"
        />
        <q-route-tab
          name="ingestLogsFromOtel"
          :to="{
            name: 'ingestLogsFromOtel',
            query: {
              org_identifier: store.state.selectedOrganization.identifier,
            },
          }"
          :icon="'img:' + getImageURL('images/ingestion/otlp.svg')"
          label="OTEL Collector"
          content-class="tab_content"
        />
        <q-route-tab
          v-if="showSyslog"
          name="syslogNg"
          :to="{
            name: 'syslogNg',
            query: {
              org_identifier: store.state.selectedOrganization.identifier,
            },
          }"
          icon="plagiarism"
          label="Syslog-ng"
          content-class="tab_content"
        />
        <q-route-tab
          v-if="showSyslog"
          name="syslog"
          :to="{
            name: 'syslog',
            query: {
              org_identifier: store.state.selectedOrganization.identifier,
            },
          }"
          icon="plagiarism"
          label="Syslog"
          content-class="tab_content"
        />
      </q-tabs>
    </template>

    <template v-slot:after>
      <router-view
        :title="ingestiontabs"
        :currOrgIdentifier="currOrgIdentifier"
        :currUserEmail="currentUserEmail"
        @copy-to-clipboard-fn="copyToClipboardFn"
      >
      </router-view>
    </template>
  </q-splitter>
</template>

<script lang="ts">
// @ts-ignore
import { defineComponent, ref, onBeforeMount, computed, onUpdated } from "vue";
import { useI18n } from "vue-i18n";
import { useStore } from "vuex";
import { useRouter } from "vue-router";
import { copyToClipboard, useQuasar } from "quasar";
import config from "../../../aws-exports";
import segment from "@/services/segment_analytics";
import { getImageURL, verifyOrganizationStatus } from "@/utils/zincutils";

export default defineComponent({
  name: "IngestLogs",
  props: {
    currOrgIdentifier: {
      type: String,
      default: "",
    },
  },
  setup() {
    const { t } = useI18n();
    const store = useStore();
    const q = useQuasar();
    const router: any = useRouter();
    const rowData: any = ref({});
    const confirmUpdate = ref<boolean>(false);
    const ingestiontabs = ref("");
    const currentOrgIdentifier: any = ref(
      store.state.selectedOrganization.identifier
    );
    const ingestRoutes = [
      "curl",
      "fluentbit",
      "fluentd",
      "kinesisfirehose",
      "vector",
      "syslog",
      "gcpLogs",
      "syslogNg",
    ];

    onBeforeMount(() => {
      if (ingestRoutes.includes(router.currentRoute.value.name)) {
        router.push({
          name: router.currentRoute.value.name,
          query: {
            org_identifier: store.state.selectedOrganization.identifier,
          },
        });
        return;
      }
      if (router.currentRoute.value.name === "ingestLogs") {
        router.push({
          name: "curl",
          query: {
            org_identifier: store.state.selectedOrganization.identifier,
          },
        });
        return;
      }
    });
    onUpdated(() => {
      if (router.currentRoute.value.name === "ingestLogs") {
        router.push({
          name: "curl",
          query: {
            org_identifier: store.state.selectedOrganization.identifier,
          },
        });
        return;
      }
    });

    const copyToClipboardFn = (content: any) => {
      copyToClipboard(content.innerText)
        .then(() => {
          q.notify({
            type: "positive",
            message: "Content Copied Successfully!",
            timeout: 5000,
          });
        })
        .catch(() => {
          q.notify({
            type: "negative",
            message: "Error while copy content.",
            timeout: 5000,
          });
        });

      segment.track("Button Click", {
        button: "Copy to Clipboard",
        ingestion: router.currentRoute.value.name,
        user_org: store.state.selectedOrganization.identifier,
        user_id: store.state.userInfo.email,
        page: "Ingestion",
      });
    };

    const showUpdateDialogFn = () => {
      confirmUpdate.value = true;
    };

    const showSyslog = computed(() => {
      return config.isCloud !== "true";
    });

    const showCloudIngestionOptions = computed(() => {
      return config.isCloud === "true";
    });

    return {
      t,
      store,
      router,
      config,
      rowData,
      splitterModel: ref(250),
      currentUserEmail: store.state.userInfo.email,
      currentOrgIdentifier,
      copyToClipboardFn,
      showUpdateDialogFn,
      confirmUpdate,
      getImageURL,
      verifyOrganizationStatus,
      ingestiontabs,
      showSyslog,
      showCloudIngestionOptions,
    };
  },
});
</script>

<style scoped lang="scss">
.ingestionPage {
  padding: 1.5rem 0 0;
  .head {
    padding-bottom: 1rem;
  }
  .q-tabs {
    &--vertical {
      margin: 1.5rem 1rem 0 1rem;
      .q-tab {
        justify-content: flex-start;
        padding: 0 0.6rem 0 0.6rem;
        border-radius: 0.5rem;
        margin-bottom: 0.5rem;
        text-transform: capitalize;

        &__content.tab_content {
          .q-tab {
            &__icon + &__label {
              padding-left: 0.875rem;
              font-weight: 600;
            }
          }
        }
        &--active {
          color: black;
          background-color: $accent;
        }
      }
    }
  }
}
</style>
<style lang="scss">
.ingestionPage {
  .q-tab-panel {
    padding: 0 !important;
    .tab_content {
      .q-tab__label {
        overflow: hidden !important;
        text-overflow: ellipsis !important;
        white-space: nowrap !important;
      }
    }
  }

  .q-icon > img {
    height: auto !important;
  }
}
</style>
