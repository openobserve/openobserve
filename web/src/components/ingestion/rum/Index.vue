<!-- Copyright 2023 OpenObserve Inc.

This program is free software: you can redistribute it and/or modify
it under the terms of the GNU Affero General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.

This program is distributed in the hope that it will be useful
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU Affero General Public License for more details.

You should have received a copy of the GNU Affero General Public License
along with this program.  If not, see <http://www.gnu.org/licenses/>.
-->

<!-- eslint-disable vue/x-invalid-end-tag -->
<template>
  <q-splitter
    v-model="splitterModel"
    unit="px"
    style="min-height: calc(100vh - 130px)"
  >
    <template v-slot:before>
      <q-tabs
        v-model="rumtabs"
        indicator-color="transparent"
        inline-label
        vertical
      >
        <q-route-tab
          name="rumWebTab"
          :to="{
            name: 'frontendMonitoring',
            query: {
              org_identifier: store.state.selectedOrganization.identifier,
            },
          }"
          icon="web"
          :label="t('common.browser')"
          content-class="tab_content"
        />
      </q-tabs>
    </template>

    <template v-slot:after>
      <router-view
        title="Web"
        :currOrgIdentifier="currOrgIdentifier"
        :currUserEmail="currentUserEmail"
        @copy-to-clipboard-fn="copyToClipboardFn"
      >
      </router-view>
    </template>
  </q-splitter>
</template>

<script lang="ts">
// @ts-ignore
import { defineComponent, ref, onMounted } from "vue";
import { useI18n } from "vue-i18n";
import { useStore } from "vuex";
import { useRouter } from "vue-router";
import { copyToClipboard, useQuasar } from "quasar";
// import { config } from "../constants/config";
import config from "../../../aws-exports";
import segment from "@/services/segment_analytics";
import { getImageURL, verifyOrganizationStatus } from "@/utils/zincutils";

export default defineComponent({
  name: "IngestRum",
  components: {},
  data() {
    return {
      rumtabs: "rumWebTab",
    };
  },
  props: {
    currOrgIdentifier: {
      type: String,
      default: "",
    },
  },
  setup() {
    const { t } = useI18n();
    const store = useStore();
    const q = useQuasar();
    const router: any = useRouter();
    const rowData: any = ref({});
    const confirmUpdate = ref<boolean>(false);

    onMounted(() => {
      const ingestRoutes = ["frontendMonitoring"];

      if (ingestRoutes.includes(router.currentRoute.value.name)) {
        router.push({
          name: router.currentRoute.value.name,
          query: {
            org_identifier: store.state.selectedOrganization.identifier,
          },
        });
        return;
      }
      if (router.currentRoute.value.name === "rumMonitoring") {
        router.push({
          name: "frontendMonitoring",
          query: {
            org_identifier: store.state.selectedOrganization.identifier,
          },
        });
        return;
      }
    });

    const copyToClipboardFn = (content: any) => {
      copyToClipboard(content.innerText)
        .then(() => {
          q.notify({
            type: "positive",
            message: "Content Copied Successfully!",
            timeout: 5000,
          });
        })
        .catch(() => {
          q.notify({
            type: "negative",
            message: "Error while copy content.",
            timeout: 5000,
          });
        });

      segment.track("Button Click", {
        button: "Copy to Clipboard",
        ingestion: router.currentRoute.value.name,
        user_org: store.state.selectedOrganization.identifier,
        user_id: store.state.userInfo.email,
        page: "RUM Ingestion",
      });
    };

    const showUpdateDialogFn = () => {
      confirmUpdate.value = true;
    };

    return {
      t,
      store,
      router,
      config,
      rowData,
      splitterModel: ref(200),
      currentUserEmail: store.state.userInfo.email,
      copyToClipboardFn,
      showUpdateDialogFn,
      confirmUpdate,
      getImageURL,
      verifyOrganizationStatus,
    };
  },
});
</script>

<style scoped lang="scss">
.ingestionPage {
  padding: 1.5rem 0 0;
  .head {
    padding-bottom: 1rem;
  }
  .q-tabs {
    &--vertical {
      margin: 1.5rem 1rem 0 1rem;
      .q-tab {
        justify-content: flex-start;
        padding: 0 0.6rem 0 0.6rem;
        border-radius: 0.5rem;
        margin-bottom: 0.5rem;
        text-transform: capitalize;

        &__content.tab_content {
          .q-tab {
            &__icon + &__label {
              padding-left: 0.875rem;
              font-weight: 600;
            }
          }
        }
        &--active {
          color: black;
          background-color: $accent;
        }
      }
    }
  }
}
</style>
