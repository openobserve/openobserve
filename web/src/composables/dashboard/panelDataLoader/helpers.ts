// Copyright 2023 OpenObserve Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.

import { useStore } from "vuex";
import { isWebSocketEnabled, isStreamingEnabled } from "@/utils/zincutils";
import type { PanelState } from "./types";
import { PANEL_DATA_LOADER_DEBOUNCE_TIME } from "./constants";

export class PanelHelpers {
  private store = useStore();

  constructor(
    private state: PanelState,
    private panelSchema: any,
  ) {}

  shouldFetchAnnotations() {
    return [
      "area",
      "area-stacked",
      "bar",
      "h-bar",
      "line",
      "scatter",
      "stacked",
      "h-stacked",
    ].includes(this.panelSchema.value.type);
  }

  hasAtLeastOneQuery() {
    return this.panelSchema.value.queries?.some((q: any) => q?.query);
  }

  waitForTimeout(signal: AbortSignal) {
    return new Promise<void>((resolve, reject) => {
      const timeoutId = setTimeout(resolve, PANEL_DATA_LOADER_DEBOUNCE_TIME);

      signal.addEventListener("abort", () => {
        clearTimeout(timeoutId);
        reject(new Error("Aborted waiting for loading"));
      });
    });
  }

  waitForThePanelToBecomeVisible(signal: any, isVisible: any, forceLoad: any) {
    return new Promise<void>((resolve, reject) => {
      if (forceLoad.value == true) {
        resolve();
        return;
      }
      
      if (isVisible.value) {
        resolve();
        return;
      }

      const stopWatching = () => {}; // This will be implemented in main composable
      
      signal.addEventListener("abort", () => {
        stopWatching();
        reject(new Error("Aborted waiting for loading"));
      });
    });
  }

  addTraceId(traceId: string) {
    if (this.state.searchRequestTraceIds.includes(traceId)) {
      return;
    }
    this.state.searchRequestTraceIds = [...this.state.searchRequestTraceIds, traceId];
  }

  removeTraceId(traceId: string) {
    this.state.searchRequestTraceIds = this.state.searchRequestTraceIds.filter(
      (id: any) => id !== traceId,
    );
  }

  processApiError(error: any, type: any) {
    switch (type) {
      case "promql": {
        const errorDetailValue = error?.response?.data?.error || error?.message;
        const trimmedErrorMessage =
          errorDetailValue?.length > 300
            ? errorDetailValue.slice(0, 300) + " ..."
            : errorDetailValue;

        const errorCode =
          error?.response?.status ||
          error?.status ||
          error?.response?.data?.code ||
          "";

        this.state.errorDetail = {
          message: trimmedErrorMessage,
          code: errorCode,
        };
        break;
      }
      case "sql": {
        const errorDetailValue =
          error?.response?.data.error_detail ||
          error?.response?.data.message ||
          error?.error_detail ||
          error?.message ||
          error?.error;

        const trimmedErrorMessage =
          errorDetailValue?.length > 300
            ? errorDetailValue.slice(0, 300) + " ..."
            : errorDetailValue;

        const errorCode =
          isWebSocketEnabled(this.store.state) || isStreamingEnabled(this.store.state)
            ? error?.response?.status ||
              error?.status ||
              error?.response?.data?.code ||
              error?.code ||
              ""
            : error?.response?.status ||
              error?.response?.data?.code ||
              error?.status ||
              error?.code ||
              "";

        this.state.errorDetail = {
          message: trimmedErrorMessage,
          code: errorCode,
        };
        break;
      }
      default:
        break;
    }
  }
}
