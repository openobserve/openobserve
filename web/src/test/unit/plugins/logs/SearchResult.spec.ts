// Copyright 2023 OpenObserve Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.

import { describe, expect, it, beforeEach, vi, afterEach } from "vitest";
import { mount, flushPromises } from "@vue/test-utils";
import { installQuasar } from "../../helpers/install-quasar-plugin";
import { Dialog, Notify } from "quasar";

import Index from "@/plugins/logs/Index.vue";
import SearchResult from "@/plugins/logs/SearchResult.vue";
// import BarChart from "@/components/logBarChart.vue";
import i18n from "@/locales";
import store from "../../helpers/store";
// @ts-ignore
import { rest } from "msw";
// import "plotly.js";
import DetailTable from "@/plugins/logs/DetailTable.vue";
import router from "../../helpers/router";

const node = document.createElement("div");
node.setAttribute("id", "app");
document.body.appendChild(node);

installQuasar({
  plugins: [Dialog, Notify],
});

describe("Search Result", async () => {
  let wrapper: any;
  beforeEach(async () => {
    vi.useFakeTimers();
    wrapper = mount(Index, {
      attachTo: "#app",
      global: {
        provide: {
          store: store,
        },
        plugins: [i18n, router],
        stubs: {
          SearchBar: true,
          IndexList: true,
        },
      },
    });
    await flushPromises();
    vi.advanceTimersByTime(500);
    await flushPromises();
  });

  afterEach(() => {
    wrapper.unmount();
    vi.restoreAllMocks();
    // vi.clearAllMocks();
  });

  it("Should show the BarChart component when showHistogram is true and sqlMode is false", async () => {
    expect(
      wrapper.find('[data-test="logs-search-result-bar-chart"]').exists()
    ).toBeTruthy();
  });

  // it("Should call the 'onChartUpdate' method when the 'updated:chart' event is emitted by the BarChart component", async () => {
  //   await wrapper.findComponent(BarChart).vm.$emit("updated:chart", {
  //     start: "2000-04-08 05:13:29",
  //     end: "2000-07-06 18:53:45",
  //   });
  //   expect(
  //     wrapper.findComponent(SearchResult).emitted()["update:datetime"]
  //   ).toBeTruthy();
  // });

  //   it("Should render the q-virtual-scroll component with the correct props and call the 'onScroll' method when the 'virtual-scroll' event is emitted", () => {
  //     expect(
  //       wrapper.find('[data-test="logs-search-result-logs-table"]').html()
  //     ).toMatchSnapshot();
  //   });

  describe("When user opens log details", () => {
    let expandRowDetail: any, hits: any[], addSearchTerm: any;
    beforeEach(async () => {
      expandRowDetail = vi.spyOn(
        wrapper.findComponent(SearchResult).vm,
        "expandRowDetail"
      );
      addSearchTerm = vi.spyOn(
        wrapper.findComponent(SearchResult).vm,
        "addSearchTerm"
      );
      hits = wrapper.vm.searchObj.data.queryResults.hits;
      await wrapper
        .find(
          `[data-test="logs-search-result-detail-${wrapper.vm.searchObj.data.queryResults.hits[0]["_timestamp"]}"]`
        )
        .trigger("click");
    });
    it('Should call the "expandRowDetail" method when a row is clicked', async () => {
      expect(expandRowDetail).toHaveBeenCalledTimes(1);
      expect(wrapper.findComponent(DetailTable).text()).toContain(
        hits[0]["_timestamp"]
      );
    });

    it('Should call "navigateRowDetail" on "showNextDetail" or "showPrevDetail" event from DetailTable', async () => {
      await wrapper
        .findComponent(DetailTable)
        .find('[data-test="log-detail-next-detail-btn"]')
        .trigger("click");
      expect(wrapper.findComponent(DetailTable).text()).toContain(
        hits[1]["_timestamp"]
      );
      expect(wrapper.findComponent(DetailTable).text()).not.toContain(
        hits[0]["_timestamp"]
      );
    });

    it('Should add or remove search terms on "add:searchterm" or "remove:searchterm" event from DetailTable', () => {
      wrapper
        .findComponent(DetailTable)
        .vm.$emit("add:searchterm", "_timestamp");
      expect(addSearchTerm).toHaveBeenCalledTimes(1);

      wrapper
        .findComponent(DetailTable)
        .vm.$emit("remove:searchterm", "_timestamp");
      expect(
        wrapper.findComponent(SearchResult).emitted()["remove:searchTerm"]
      ).toBeTruthy();
    });

    it('Should execute timeboxed search on "search:timeboxed" event from DetailTable component', async () => {
      global.server.use(
        rest.get(
          `${store.state.API_ENDPOINT}/api/${store.state.selectedOrganization.identifier}/k8s_json/_around`,
          (req: any, res: any, ctx: any) => {
            return res(
              ctx.status(200),
              ctx.json({
                took: 10,
                hits: [
                  {
                    _timestamp: 1680246906641572,
                    kubernetes_annotations_kubernetes_io_psp: "eks.privileged",
                    kubernetes_container_hash:
                      "058694856476.dkr.ecr.us-west-2.amazonaws.com/ziox@sha256:3dbbb0dc1eab2d5a3b3e4a75fd87d194e8095c92d7b2b62e7cdbd07020f54589",
                    kubernetes_container_image:
                      "058694856476.dkr.ecr.us-west-2.amazonaws.com/ziox:v0.0.3",
                    kubernetes_container_name: "ziox",
                    kubernetes_docker_id:
                      "65917867b45dbe2cd429361796b5e7d42581124411bec968acfbd80ad0f163e2",
                    kubernetes_host:
                      "ip-10-2-15-197.us-east-2.compute.internal",
                    kubernetes_labels_app: "ziox",
                    kubernetes_labels_name: "ziox-querier",
                    kubernetes_labels_pod_template_hash: "595748494c",
                    kubernetes_labels_role: "querier",
                    kubernetes_namespace_name: "ziox",
                    kubernetes_pod_id: "ff96944f-ca8e-413f-a696-3236b1145482",
                    kubernetes_pod_name: "ziox-querier-595748494c-vd9zw",
                    log: '[2022-12-27T14:12:29Z INFO  zinc_enl::service::search::datafusion] search file: Bhargav_organization_29/logs/default/2022/12/27/05/7013381009471324160.parquet, need add columns: ["from"]',
                    stream: "stderr",
                  },
                ],
                total: 1,
                from: 0,
                size: 1,
                scan_size: 110,
              })
            );
          }
        )
      );
      await wrapper
        .findComponent(DetailTable)
        .find('[data-test="logs-detail-table-search-around-btn"]')
        .trigger("click");
      expect(
        wrapper.findComponent(SearchResult).emitted()["search:timeboxed"]
      ).toBeTruthy();
    });
  });
});
