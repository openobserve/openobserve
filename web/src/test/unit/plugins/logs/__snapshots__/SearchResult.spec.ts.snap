// Vitest Snapshot v1

exports[`Search Result > Should render the q-virtual-scroll component with the correct props and call the 'onScroll' method when the 'virtual-scroll' event is emitted 1`] = `
"<div class=\\"q-markup-table q-table__container q-table__card q-table--horizontal-separator q-table--no-wrap q-virtual-scroll q-virtual-scroll--vertical scroll\\" data-test=\\"logs-search-result-logs-table\\" id=\\"searchGridComponent\\" tabindex=\\"0\\" data-v-ae03aa38=\\"\\" style=\\"height: calc(100vh - 0px);\\">
  <table class=\\"q-table\\">
    <thead class=\\"thead-sticky text-left\\" data-v-ae03aa38=\\"\\">
      <tr data-v-ae03aa38=\\"\\">
        <th class=\\"table-header\\" data-v-ae03aa38=\\"\\"><span class=\\"table-head-label\\" data-v-ae03aa38=\\"\\">timestamp</span></th>
        <th class=\\"table-header\\" data-v-ae03aa38=\\"\\"><span class=\\"table-head-label\\" data-v-ae03aa38=\\"\\">source</span></th>
      </tr>
    </thead>
    <tbody class=\\"q-virtual-scroll__padding\\">
      <tr>
        <td style=\\"height: 0px; --q-virtual-scroll-item-height: 25px;\\" colspan=\\"100\\"></td>
      </tr>
    </tbody>
    <tbody class=\\"q-virtual-scroll__content\\" tabindex=\\"-1\\">
      <tr class=\\"q-tr\\" data-test=\\"logs-search-result-detail-1680246906644637\\" style=\\"cursor: pointer;\\" data-v-ae03aa38=\\"\\">
        <td class=\\"q-td \\" data-v-ae03aa38=\\"\\">
          <div data-v-ae03aa38=\\"\\"><span iskeyword=\\"false\\" text=\\"Mar 31, 2023 12:45:06.644 +05:30\\"><span title=\\"\\">Mar 31, 2023 12:45:06.644 +05:30</span></span></div>
        </td>
        <td class=\\"q-td \\" data-v-ae03aa38=\\"\\">
          <div data-v-ae03aa38=\\"\\"><span iskeyword=\\"false\\" text=\\"{&quot;_timestamp&quot;:1680246906644637,&quot;kubernetes_annotations_kubernetes_io_psp&quot;:&quot;eks.privileged&quot;,&quot;kubernetes_container_hash&quot;:&quot;058694856476.dkr.ecr.us-west-2.amazonaws.com/ziox@sha256:3dbbb0dc1eab2d5a3b3e4a75fd87d194e8095c92d7b2b62e7cdbd07020f54589&quot;,&quot;kubernetes_container_image&quot;:&quot;058694856476.dkr.ecr.us-west-2.amazonaws.com/ziox:v0.0.3&quot;,&quot;kubernetes_container_name&quot;:&quot;ziox&quot;,&quot;kubernetes_docker_id&quot;:&quot;56510ea4003f201fa7913c18d7c8dc639c576e5d1c4bbf0ead911d2d68ae8f31&quot;,&quot;kubernetes_host&quot;:&quot;ip-10-2-56-221.us-east-2.compute.internal&quot;,&quot;kubernetes_labels_app&quot;:&quot;ziox&quot;,&quot;kubernetes_labels_name&quot;:&quot;ziox-querier&quot;,&quot;kubernetes_labels_pod_template_hash&quot;:&quot;595748494c&quot;,&quot;kubernetes_labels_role&quot;:&quot;querier&quot;,&quot;kubernetes_namespace_name&quot;:&quot;ziox&quot;,&quot;kubernetes_pod_id&quot;:&quot;810e7dbc-fb09-4a8f-9108-6b8785754af9&quot;,&quot;kubernetes_pod_name&quot;:&quot;ziox-querier-595748494c-7xqlg&quot;,&quot;log&quot;:&quot;[2022-12-27T14:12:29Z INFO  zinc_enl::service::search::datafusion] search file: Bhargav_organization_29/logs/default/2022/12/23/12/7012038228660338688.parquet, need add columns: [\\\\&quot;from\\\\&quot;]&quot;,&quot;stream&quot;:&quot;stderr&quot;}\\"><span title=\\"\\">{\\"_timestamp\\":1680246906644637,\\"kubernetes_annotations_kubernetes_io_psp\\":\\"eks.privileged\\",\\"kubernetes_container_hash\\":\\"058694856476.dkr.ecr.us-west-2.amazonaws.com/ziox@sha256:3dbbb0dc1eab2d5a3b3e4a75fd87d194e8095c92d7b2b62e7cdbd07020f54589\\",\\"kubernetes_container_image\\":\\"058694856476.dkr.ecr.us-west-2.amazonaws.com/ziox:v0.0.3\\",\\"kubernetes_container_name\\":\\"ziox\\",\\"kubernetes_docker_id\\":\\"56510ea4003f201fa7913c18d7c8dc639c576e5d1c4bbf0ead911d2d68ae8f31\\",\\"kubernetes_host\\":\\"ip-10-2-56-221.us-east-2.compute.internal\\",\\"kubernetes_labels_app\\":\\"ziox\\",\\"kubernetes_labels_name\\":\\"ziox-querier\\",\\"kubernetes_labels_pod_template_hash\\":\\"595748494c\\",\\"kubernetes_labels_role\\":\\"querier\\",\\"kubernetes_namespace_name\\":\\"ziox\\",\\"kubernetes_pod_id\\":\\"810e7dbc-fb09-4a8f-9108-6b8785754af9\\",\\"kubernetes_pod_name\\":\\"ziox-querier-595748494c-7xqlg\\",\\"log\\":\\"[2022-12-27T14:12:29Z INFO  zinc_enl::service::search::datafusion] search file: Bhargav_organization_29/logs/default/2022/12/23/12/7012038228660338688.parquet, need add columns: [\\\\\\"from\\\\\\"]\\",\\"stream\\":\\"stderr\\"}</span></span></div>
        </td>
      </tr>
      <tr class=\\"q-tr\\" data-test=\\"logs-search-result-detail-1680246906644599\\" style=\\"cursor: pointer;\\" data-v-ae03aa38=\\"\\">
        <td class=\\"q-td \\" data-v-ae03aa38=\\"\\">
          <div data-v-ae03aa38=\\"\\"><span iskeyword=\\"false\\" text=\\"Mar 31, 2023 12:45:06.644 +05:30\\"><span title=\\"\\">Mar 31, 2023 12:45:06.644 +05:30</span></span></div>
        </td>
        <td class=\\"q-td \\" data-v-ae03aa38=\\"\\">
          <div data-v-ae03aa38=\\"\\"><span iskeyword=\\"false\\" text=\\"{&quot;_timestamp&quot;:1680246906644599,&quot;kubernetes_annotations_kubernetes_io_psp&quot;:&quot;eks.privileged&quot;,&quot;kubernetes_container_hash&quot;:&quot;058694856476.dkr.ecr.us-west-2.amazonaws.com/ziox@sha256:3dbbb0dc1eab2d5a3b3e4a75fd87d194e8095c92d7b2b62e7cdbd07020f54589&quot;,&quot;kubernetes_container_image&quot;:&quot;058694856476.dkr.ecr.us-west-2.amazonaws.com/ziox:v0.0.3&quot;,&quot;kubernetes_container_name&quot;:&quot;ziox&quot;,&quot;kubernetes_docker_id&quot;:&quot;56510ea4003f201fa7913c18d7c8dc639c576e5d1c4bbf0ead911d2d68ae8f31&quot;,&quot;kubernetes_host&quot;:&quot;ip-10-2-56-221.us-east-2.compute.internal&quot;,&quot;kubernetes_labels_app&quot;:&quot;ziox&quot;,&quot;kubernetes_labels_name&quot;:&quot;ziox-querier&quot;,&quot;kubernetes_labels_pod_template_hash&quot;:&quot;595748494c&quot;,&quot;kubernetes_labels_role&quot;:&quot;querier&quot;,&quot;kubernetes_namespace_name&quot;:&quot;ziox&quot;,&quot;kubernetes_pod_id&quot;:&quot;810e7dbc-fb09-4a8f-9108-6b8785754af9&quot;,&quot;kubernetes_pod_name&quot;:&quot;ziox-querier-595748494c-7xqlg&quot;,&quot;log&quot;:&quot;[2022-12-27T14:12:29Z INFO  aws_config::default_provider::credentials] provide_credentials; provider=default_chain&quot;,&quot;stream&quot;:&quot;stderr&quot;}\\"><span title=\\"\\">{\\"_timestamp\\":1680246906644599,\\"kubernetes_annotations_kubernetes_io_psp\\":\\"eks.privileged\\",\\"kubernetes_container_hash\\":\\"058694856476.dkr.ecr.us-west-2.amazonaws.com/ziox@sha256:3dbbb0dc1eab2d5a3b3e4a75fd87d194e8095c92d7b2b62e7cdbd07020f54589\\",\\"kubernetes_container_image\\":\\"058694856476.dkr.ecr.us-west-2.amazonaws.com/ziox:v0.0.3\\",\\"kubernetes_container_name\\":\\"ziox\\",\\"kubernetes_docker_id\\":\\"56510ea4003f201fa7913c18d7c8dc639c576e5d1c4bbf0ead911d2d68ae8f31\\",\\"kubernetes_host\\":\\"ip-10-2-56-221.us-east-2.compute.internal\\",\\"kubernetes_labels_app\\":\\"ziox\\",\\"kubernetes_labels_name\\":\\"ziox-querier\\",\\"kubernetes_labels_pod_template_hash\\":\\"595748494c\\",\\"kubernetes_labels_role\\":\\"querier\\",\\"kubernetes_namespace_name\\":\\"ziox\\",\\"kubernetes_pod_id\\":\\"810e7dbc-fb09-4a8f-9108-6b8785754af9\\",\\"kubernetes_pod_name\\":\\"ziox-querier-595748494c-7xqlg\\",\\"log\\":\\"[2022-12-27T14:12:29Z INFO  aws_config::default_provider::credentials] provide_credentials; provider=default_chain\\",\\"stream\\":\\"stderr\\"}</span></span></div>
        </td>
      </tr>
    </tbody>
    <tbody class=\\"q-virtual-scroll__padding\\">
      <tr>
        <td style=\\"height: 0px; --q-virtual-scroll-item-height: 25px;\\" colspan=\\"100\\"></td>
      </tr>
    </tbody>
  </table>
</div>"
`;
